File: autocare-backend/src/main/java/com/pobitra/autocare/service/FeedbackService.java
package com.pobitra.autocare.service;

import com.pobitra.autocare.dto.FeedbackRequestDTO;
import com.pobitra.autocare.dto.FeedbackResponseDTO;

import java.util.List;

public interface FeedbackService {

    FeedbackResponseDTO createFeedback(FeedbackRequestDTO dto);

    List<FeedbackResponseDTO> getAllFeedbacks();

    FeedbackResponseDTO getFeedbackById(Long id);

    List<FeedbackResponseDTO> getFeedbackByRating(Integer rating);

    List<FeedbackResponseDTO> getFeedbackByCustomer(Long customerId);

    void deleteFeedback(Long id);

}

File: autocare-frontend/src/services/feedbackService.js
import api from "./api";

const feedbackService = {

    getAllFeedback() {
        return api.get("/feedback");
    },

    getFeedbackById(id) {
        return api.get(`/feedback/${id}`);
    },

    createFeedback(feedback) {
        return api.post("/feedback", feedback);
    },

    updateFeedback(id, feedback) {
        return api.put(`/feedback/${id}`, feedback);
    },

    deleteFeedback(id) {
        return api.delete(`/feedback/${id}`);
    },

    getFeedbackByCustomer(customerId) {
        return api.get(`/feedback/customer/${customerId}`);
    },

    getFeedbackByRating(rating) {
        return api.get(`/feedback/rating/${rating}`);
    }

};

export default feedbackService;